nickel-mode.el

```diff
@@ -0,0 +1,103 @@
+;;; nickel-mode.el --- A major mode for editing Nickel source code -*-lexical-binding: t-*-
+
+;; Copyright (c) Tweag Holding and its affiliates.
+
+;; Version: 0.1
+;; Author: The Nickel Team (nickel-lang@tweag.io)
+;; Url: https://github.com/nickel-lang/nickel-mode
+;; Created: 7 March 2023
+;; Keywords: languages configuration-language configuration nickel infrastructure
+;; Homepage: https://nickel-lang.org/
+;; Package-Requires: ((emacs "24.1"))
+
+;; This file is distributed under the terms of MIT license.
+
+;;; Commentary:
+
+;; This package implements a major-mode for editing Nickel source code.
+
+
+;;; Code:
+
+(defgroup nickel nil
+  "Major mode for editing Nickel source code."
+  :group 'languages
+  :link '(url-link :tag "Site" "https://nickel-lang.org/")
+  :link '(url-link :tag "Repository" "https://github.com/nickel-lang/nickel-mode"))
+
+(defvar nickel-mode-keywords
+  (regexp-opt '("if" "then" "else" "forall" "in" "let"
+                "rec" "match" "fun" "import" "merge"
+                "default" "doc" "optional" "priority" "force" "not_exported")
+	      'word))
+
+(defvar nickel-mode-constants
+  (regexp-opt '("true" "false" "null")
+	     'word))
+
+(defvar nickel-mode-types
+  (regexp-opt '("Dyn" "Num" "Str" "Array")
+	      'word))
+
+(defvar nickel-mode-primops
+  (regexp-opt '("%typeof%" "%assume%" "%array_lazy_assume%" "%dictionary_assume%"
+	        "%blame%" "%chng_pol%" "%polarity%" "%go_dom%" "%go_codom%" "%go_field%"
+	        "%go_array%" "%go_dict%" "%seal%" "%unseal%" "%embed%" "%record_map%"
+	        "%record_insert%" "%record_remove%" "%record_empty_with_tail%" "%record_seal_tail%"
+	        "%record_unseal_tail%" "%seq%" "%deep_seq%" "%force%" "%head%" "%tail%" "%length%"
+	        "%fields%" "%values%" "%pow%" "%trace%" "%has_field%" "%map%" "%elem_at%" "%generate%"
+	        "%rec_force%" "%rec_default%")
+	      'symbol))
```

Comment:
```suggestion
(defconst nickel-mode-keywords
  (regexp-opt '("if" "then" "else" "forall" "in" "let"
                "rec" "match" "fun" "import" "merge"
                "default" "doc" "optional" "priority" "force" "not_exported")
              'symbols))

(defconst nickel-mode-constants
  (regexp-opt '("true" "false" "null")
              'symbols))

(defconst nickel-mode-types
  (regexp-opt '("Dyn" "Num" "Str" "Array")
              'symbols))

(defconst nickel-mode-primops
  (regexp-opt '("%typeof%" "%assume%" "%array_lazy_assume%" "%dictionary_assume%"
                "%blame%" "%chng_pol%" "%polarity%" "%go_dom%" "%go_codom%" "%go_field%"
                "%go_array%" "%go_dict%" "%seal%" "%unseal%" "%embed%" "%record_map%"
                "%record_insert%" "%record_remove%" "%record_empty_with_tail%" "%record_seal_tail%"
                "%record_unseal_tail%" "%seq%" "%deep_seq%" "%force%" "%head%" "%tail%" "%length%"
                "%fields%" "%values%" "%pow%" "%trace%" "%has_field%" "%map%" "%elem_at%" "%generate%"
                "%rec_force%" "%rec_default%")
              'symbols))
```

Reply:
Oops, that probably changed the tabs too, but I'd recommend sticking with spaces. I'll add another suggestion to make this automatic.

Reply:
Thanks 🙂